orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -0,0 +1,166 @@
+import retrofit.RetrofitError;
+
+public class MonitoredDeployBaseTask implements RetryableTask {
+  private static final int MAX_RETRY_COUNT = 3;
+  protected final Logger log = LoggerFactory.getLogger(getClass());
+  protected DeploymentMonitorDefinition monitorDefinition;
+  protected Stage stage;
+  protected Registry registry;
+
+  private DeploymentMonitorServiceProvider deploymentMonitorServiceProvider;
+
+  MonitoredDeployBaseTask(
+      DeploymentMonitorServiceProvider deploymentMonitorServiceProvider, Registry registry) {
+    this.deploymentMonitorServiceProvider = deploymentMonitorServiceProvider;
+    this.registry = registry;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.MINUTES.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    // TODO(mvulfson): Use DeploymentMonitorDefinition
+    return TimeUnit.MINUTES.toMillis(30);
+  }
+
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    TaskResult result;
+    MonitoredDeployStageData context = stage.mapTo(MonitoredDeployStageData.class);
+
+    try {
+      this.stage = stage;
+      this.monitorDefinition =
+          deploymentMonitorServiceProvider.getDefinitionById(
+              context.getDeploymentMonitor().getId());
+
+      result = executeInternal();
+    } catch (RetrofitError e) {
+      registry
+          .counter("deploymentMonitor.errors", "monitorId", monitorDefinition.getId())
+          .increment();
+
+      int currentRetryCount = context.getDeployMonitorHttpRetryCount();
+
+      if (currentRetryCount >= MAX_RETRY_COUNT) {
+        if (monitorDefinition.isFailOnError()) {
+          registry
+              .counter("deploymentMonitor.fatalErrors", "monitorId", monitorDefinition.getId())
+              .increment();
+
+          log.error(
+              "Failed to get valid response for {} from deployment monitor {}, aborting because the monitor is marked with failOnError",
+              getClass().getSimpleName(),
+              monitorDefinition);
+
+          result =
+              TaskResult.builder(ExecutionStatus.TERMINAL)
+                  // TODO(mvulfson)
+                  // .context()
+                  .build();
+        } else {
+          log.warn(
+              "Failed to get valid response for {} from deployment monitor {}, ignoring failure",
+              getClass().getSimpleName(),
+              monitorDefinition);
+
+          result =
+              TaskResult.builder(ExecutionStatus.SUCCEEDED)
+                  // TODO(mvulfson)
+                  // .context()
+                  .build();
+        }
+      } else {
+        // Retry
+        result =
+            TaskResult.builder(ExecutionStatus.RUNNING)
+                .context("deployMonitorHttpRetryCount", ++currentRetryCount)
+                .build();
+      }
+    } catch (Exception e) {
+      log.error("Exception while executing {}, aborting deployment", getClass().getSimpleName(), e);
```

Comment:
might as well not have this catch block if we just catch, log and rethrow, no? Something will catch and log it further up

Reply:
result is not set here, so I guess we can end up returning `null`?